src/cljs/main/broadfcui/page/method_repo/create_method.cljs

```diff
@@ -94,56 +108,71 @@
+              [comps/Checkbox {:ref "redact-checkbox"
+                               :label (str "Redact Snapshot " (:snapshotId info))}]])
+
+           [comps/ErrorViewer {:error (:upload-error @state)}]
+           (style/create-validation-error-message (:validation-errors @state))])
          :ok-button (react/create-element
                      [comps/Button {:ref "ok-button"
-                                    :text (:ok-text defaults)
-                                    :onClick #(this :create-method)}])}]))
+                                    :text (:ok-text info)
+                                    :onClick #(this :-create-method)}])}]))
    :component-did-mount
    (fn [{:keys [state refs]}]
      ((@refs "wdl-editor") :add-listener "change"
       #(swap! state assoc :undo-history
               (js->clj ((@refs "wdl-editor") :call-method "historySize")))))
-   :set-wdl-text
+   :-set-wdl-text
    (fn [{:keys [refs]} text]
      ((@refs "wdl-editor") :call-method "setValue" text))
-   :create-method
-   (fn [{:keys [props state refs]}]
+   :-create-method
+   (fn [{:keys [state locals refs this]}]
      (let [[namespace name & fails] (input/get-and-validate refs "namespace" "name")
            [synopsis documentation] (common/get-text refs "synopsis" "documentation")
            wdl ((@refs "wdl-editor") :call-method "getValue")
            fails (or fails (when (clojure.string/blank? wdl) ["Please enter the WDL payload"]))]
        (swap! state assoc :validation-errors fails)
        (when-not fails
-         (swap! state assoc :uploading? true)
-         (endpoints/call-ajax-orch
-          {:endpoint endpoints/post-method
-           :payload {:namespace namespace
-                     :name name
-                     :synopsis synopsis
-                     :documentation documentation
-                     :payload wdl
-                     :entityType "Workflow"}
-           :headers utils/content-type=json
-           :on-done
-           (fn [{:keys [success? get-parsed-response]}]
-             (swap! state dissoc :uploading?)
-             (if success?
-               (do
-                 (modal/pop-modal)
-                 (let [response (get-parsed-response)
-                       {:keys [namespace name snapshotId]} response
-                       id {:namespace namespace :name name :snapshot-id snapshotId}]
-                   ((:on-created props) :method id)))
-               (swap! state assoc :upload-error (get-parsed-response false))))}))))})
+         (swap! state assoc :banner "Uploading...")
+         (if (:edit-mode? (:info @locals))
+           (let [{:keys [snapshotId]} (:info @locals)
+                 redact? ((@refs "redact-checkbox") :checked?)]
+             (endpoints/call-ajax-orch
+              {:endpoint (endpoints/create-new-method-snapshot namespace name snapshotId redact?)
+               :payload (assoc (utils/restructure synopsis documentation) :payload wdl)
```

Comment:
brainstorming: if it would help to simplify the UI code, you *could* send the namespace/name/entityType to the `create-new-method-snapshot` endpoint; the endpoint will ignore those values. It would be pretty confusing for anybody reading the payloads, or this code, though.